nestkernel/simulation_manager.cpp

```diff
@@ -443,13 +499,17 @@ nest::SimulationManager::simulate( Time const& t )
       "is called repeatedly with simulation times that are not multiples of "
       "the minimal delay." );
 
-  resume_( num_active_nodes );
+  resume_();
+}
 
+void
+nest::SimulationManager::cleanup()
+{
   finalize_simulation_();
 }
 
 void
-nest::SimulationManager::resume_( size_t num_active_nodes )
+nest::SimulationManager::resume_()
 {
   assert( kernel().is_initialized() and not inconsistent_state_ );
 
```

Comment:
The next section of code is lengthy and thus puts together a status message. Should that be delegated to a method, so that the flow of important logic becomes more transparent here?